pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -613,6 +614,16 @@ func getTaskRunsStatus(pr *v1beta1.PipelineRun, state []*resources.ResolvedPipel
 	return status
 }
 
+func getSkippedTasks(pr *v1beta1.PipelineRun, state []*resources.ResolvedPipelineRunTask, d *dag.Graph) []v1beta1.SkippedTask {
+	skipped := []v1beta1.SkippedTask{}
+	for _, rprt := range state {
+		if rprt.Skip(state, d) {
```

Comment:
Nice :)

The `Skip` method will be true in a number of cases:
- when expression is failed
- condition is failed
- a task is not reachable via the DAG (because of a previous skip)
- a task failed so no new task is scheduled

It might we worth adding to the docs this information. It can be in a PR on top.